src/makeCalculator.js

```diff
@@ -4,7 +4,33 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+    operate(action, number) {
+      this.result = action(this.result, number);
+
+      return this;
+    },
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+    add(currentResult, number) {
+      return currentResult + number;
+    },
+    subtract(currentResult, number) {
+      return currentResult - number;
+    },
+    multiply(currentResult, number) {
+      return currentResult * number;
+    },
+    divide(currentResult, number) {
+      return currentResult / number;
+    },
```

Comment:
It is necessary to add a check for 0, since it is not possible to divide by 0 according to the rules